### PR TITLE
feat: fw-276 claim redpacket

### PR DESCRIPTION
**Pull request by UncleBill**
https://mask.atlassian.net/browse/FW-276
MF-0000

### CODE REVIEW

File: packages/polyfills/types/global.d.ts [hunk 1]
```diff
@@ -11,3 +11,5 @@ type LiteralUnion<
     : U extends number ? number
     : never,
 > = U | (T & Nothing)
+
+type HexString = `0x${string}`
```
Comment (Jack-Works) [inline packages/polyfills/types/global.d.ts:15]: I'll not suggest you to add it in the polyfills package, but you can go if you really want.

File: packages/plugins/RedPacket/src/SiteAdaptor/RedPacket/index.tsx [hunk 2]
```diff
@@ -273,9 +218,27 @@ export const RedPacket = memo(function RedPacket({ payload }: RedPacketProps) {
...
 
+    const cover = getRedPacketCover({
+        theme: 'lucky-flower',
+        symbol: token.symbol,
+        shares: payload.shares,
+        amount: formatBalance(payload.total, token.decimals, { significant: 2 }),
+        from: `@${payload.sender.name}`,
+        message: payload.sender.message,
+        remainingAmount: payload.total_remaining!,
+        remainingShares: payload.total_remaining!,
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/RedPacket/index.tsx]: The field value is wrong.

File: packages/plugins/RedPacket/src/SiteAdaptor/RedPacketNft.tsx [hunk 3]
```diff
@@ -312,21 +330,41 @@ export function RedPacketNft({ payload }: RedPacketNftProps) {
...
 
+    const cover = getRedPacketCover({
+        symbol: asset?.metadata?.symbol!,
+        theme: 'lucky_flower',
+        shares: availability.totalAmount,
+        amount: availability.totalAmount,
+        from: payload.senderName,
+        message: payload.message,
+        remainingAmount: availability.remaining,
+        remainingShares: availability.remaining!,
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/RedPacketNft.tsx]: A wrong field value.

File: packages/plugins/RedPacket/src/SiteAdaptor/Requirements.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+                            <ResultIcon className={classes.state} size={18} result={condition.value} />
+                        </ListItem>
+                    ))
+            }
+            if (status.type === 'nftOwned') {
+                const collectionNames = status.payload.map((x) => x.collectionName).join(', ')
+                return (
+                    <ListItem className={classes.item} key={status.type}>
+                        <Icons.FireflyNFT className={classes.icon} size={16} />
+                        <Typography className={classes.text}>NFT Holder of {collectionNames}</Typography>
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/Requirements.tsx:124]: Locale

File: packages/plugins/RedPacket/src/SiteAdaptor/Requirements.tsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    const requirements = useMemo(() => {
+        return statusList.flatMap((status) => {
+            if (status.type === 'profileFollow') {
+                const payload = status.payload.filter((x) => x.platform === 'lens')
+                const handles = payload.map((x) => `@${x.handle}`).join(', ')
+                return (
+                    <ListItem className={classes.item} key={status.type}>
+                        <Icons.UserPlus className={classes.icon} size={16} />
+                        <Typography className={classes.text}>
+                            Follow {handles} on {payload[0].platform}
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/Requirements.tsx]: Locale

File: packages/plugins/RedPacket/src/SiteAdaptor/utils/getRedPacketCover.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    symbol: string
+    shares: number | string
+    amount: number | string
+    from: string
+    message: string
+    remainingAmount: number | string
+    remainingShares: number | string
+}
+
+export function getRedPacketCover(options: Options) {
```
Comment (nuanyang233) [inline packages/plugins/RedPacket/src/SiteAdaptor/utils/getRedPacketCover.ts:14]: We can handle the fetching of cover in the FireflyRedpacket api

Comment (UncleBill) [reply packages/plugins/RedPacket/src/SiteAdaptor/utils/getRedPacketCover.ts:14]: This is not fetching but constructing the url

**Pull request merged**